src/main/java/com/cgdpd/library/entity/TrackingStatus.java

```diff
@@ -0,0 +1,67 @@
+package com.cgdpd.library.entity;
+
+import java.util.EnumSet;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+public enum TrackingStatus {
+    AVAILABLE("Available"),
+    ON_HOLD("On Hold"),
+    CHECKED_OUT("Checked Out"),
+    LOST("Lost"),
+    RETIRED("Retired"),
+    REFERENCE("Reference");
+
+    private final String status;
+    private static final Map<TrackingStatus, Set<TrackingStatus>> nextStatuses = new HashMap<>();
+
+    TrackingStatus(String status) {
+        this.status = status;
+    }
+
+    public String getStatus() {
+        return status;
+    }
+
+    public Map<TrackingStatus, Set<TrackingStatus>> getNextStatuses() {
+        return nextStatuses;
+    }
+
+    public void addNextStatus(TrackingStatus nextStatus) {
+        nextStatuses.computeIfAbsent(this, k -> EnumSet.noneOf(TrackingStatus.class))
+              .add(nextStatus);
+    }
+
+    public boolean isNextStatusAvailable(TrackingStatus nextStatus) {
+        return nextStatuses.getOrDefault(this, EnumSet.noneOf(TrackingStatus.class))
```

Comment:
We should not return default value here, if we don't have the enum value in the map then it is a programmer error
```suggestion
        return allowedTransitions.get(this).contains(nextStatus);
```